packages/components/src/components/sidenavigation/sidenavigation.styles.ts

```diff
@@ -70,6 +71,16 @@ const styles = css`
   :host::part(vertical-divider-button) {
     z-index: var(--mdc-sidenavigation-vertical-divider-button-z-index);
   }
+
+  :host([variant='flexible-on-hover'])::part(vertical-divider),
+  :host([variant='flexible-on-hover'])::part(vertical-divider-button) {
+    opacity: 0;
+  }
+
+  :host([data-grabber-visible][variant='flexible-on-hover'])::part(vertical-divider),
+  :host([data-grabber-visible][variant='flexible-on-hover'])::part(vertical-divider-button) {
```

Comment:
it is really just nitpicking but for better readability you should order the selectors from the more generic to the more specific


```suggestion
  :host([variant='flexible-on-hover'][data-grabber-visible])::part(vertical-divider),
  :host([variant='flexible-on-hover'][data-grabber-visible])::part(vertical-divider-button) {
```